src/app/modules/auth/auth.middlewares.ts

```diff
@@ -92,3 +94,51 @@ export const validateVerifyOtpParams = celebrate({
       .message('Please enter a valid OTP'),
   }),
 })
+
+/**
+ * Middleware that only allows users with a valid bearer token to pass through to the next handler
+ */
+export const authenticateApiKey: ControllerHandler = (req, res, next) => {
+  const authorizationHeader = req.headers.authorization
+  if (!authorizationHeader) {
+    return res
+      .status(StatusCodes.UNAUTHORIZED)
+      .json({ message: 'Authorisation header is missing' })
+  }
+  const [bearerString, apiKey] = authorizationHeader.split(BEARER_SEPARATOR)
+  if (bearerString !== BEARER_STRING) {
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'Invalid authorisation header format' })
+  }
+  const splitApiKey = apiKey.split(API_KEY_SEPARATOR)
+  if (splitApiKey.length !== 4) {
+    return res
+      .status(StatusCodes.BAD_REQUEST)
+      .json({ message: 'Invalid API key format' })
+  }
+  const [apiEnv, apiVersion, userId, token] = splitApiKey
+  logger.info({
+    message: 'User attempting to authenticate using API key',
+    meta: {
+      action: 'authenticateApiKey',
+      apiEnv,
+      apiVersion,
+      user: userId,
+    },
+  })
+  return getUserByApiKey(userId, token)
+    .map((user) => {
+      if (!user) {
+        return res
+          .status(StatusCodes.UNAUTHORIZED)
+          .json({ message: 'Invalid API key' })
+      }
+      req.session.user = { _id: user.id }
```

Comment:
Just a note on this: this works because the session middleware is currently running for all routes served by express, and that is sort of wrong. 

We should activate the session middleware only for routes which require it (typically under the API router), and for the new "external" APIs, our auth middleware should take responsibility to setup a compatible session object. Something as small as:

```suggestion
if (!req.session) req.session = {}
req.session.user = { _id: user.id }
```